File: advcake/brandshop/get_product.py
import random
from get_sale import get_sale
from get_prices import get_prices
from get_transliterate import get_transliterate
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options
import undetected_chromedriver


def get_product(link, gender, idx):
    global category
    try:
        option = Options()
        option.headless = True
	options.add_argument("--headless=new")
	driver = undetected_chromedriver.Chrome(options=option)
        driver.get(link)
        try:
            if gender != 'women': gender = 'Мужской'
            else: gender = 'Женский'
            price = get_prices(driver.find_element(By.CLASS_NAME, 'product-order__price_new'))
            oldprice = get_prices(driver.find_element(By.CLASS_NAME, 'product-order__price_old'))
            sale = get_sale(driver.find_element(By.CLASS_NAME, 'product-order__price-discount'))
            brand = driver.find_element(By.CLASS_NAME, 'product-page__header').text.upper()
            name = driver.find_element(By.CLASS_NAME, 'product-page__subheader').text.strip() + ' ' + brand
            category = driver.find_elements(By.CLASS_NAME, 'breadcrumbs__item')[2].get_attribute('textContent').strip()
            subcategory = driver.find_elements(By.CLASS_NAME, 'breadcrumbs__item')[3].get_attribute('textContent').strip()
            color = driver.find_elements(By.CLASS_NAME, 'tooltip')[0].get_attribute('textContent').lower().replace('ё', 'е').replace('в наличии: онлайн, офлайн', '').replace('в наличии: онлайн', '')
            if color == '': color = driver.find_elements(By.CLASS_NAME, 'tooltip')[1].get_attribute('textContent').lower().replace('ё', 'е').replace('в наличии: онлайн, офлайн', '').replace('в наличии: онлайн', '')
            if color == '': color = driver.find_elements(By.CLASS_NAME, 'tooltip')[2].get_attribute('textContent').lower().replace('ё', 'е').replace('в наличии: онлайн, офлайн', '').replace('в наличии: онлайн', '')
            if color == '': color = False
            info = []
            country = False
            structure = False
            for idx, param in enumerate(driver.find_element(By.CLASS_NAME, 'product-menu__content').find_elements(By.TAG_NAME, 'li')):
                if idx == 0: structure = param.get_attribute('textContent').replace('Материал: ', '').split('; ')
                elif idx == 1: country = param.get_attribute('textContent').replace('Страна-производитель: ', '').upper()
                else: info.append(param.get_attribute('textContent'))
            sizes = []
            for size in driver.find_elements(By.CLASS_NAME, 'product-plate__item'):
                sizes.append(size.get_attribute('textContent').replace('В наличии: Онлайн, Офлайн', '').replace('В наличии: Онлайн', '').replace(' EU', '').strip())
            images = []
            for img in driver.find_elements(By.CLASS_NAME, 'zoom-region'):
                images.append(img.find_element(By.TAG_NAME, 'div').find_element(By.TAG_NAME, 'img').get_attribute('src'))
            print(idx, 'PRODUCT - DONE!!!')
        except Exception as ex:
            print(ex)
            return
        finally:
            if category == 'Верхняя одежда': category = 'Одежда'
            driver.quit()
        return {
            'id': round(random.uniform(1000000000, 9999999999)),
            'age': 'Взрослый',
            'benefit': oldprice - price,
            'brand': brand,
            'brandCountry': False,
            'brandCountry_t': False,
            'category': category,
            'category_t': get_transliterate(category),
            'color': color,
            'color_t': get_transliterate(color),
            'country': country,
            'country_t': get_transliterate(country),
            'delivery': ['ru', 'rb', 'kz'],
            'deliveryPrice': 350,
            'description': False,
            'gender': gender,
            'name': name,
            'info': info,
            'installment': True,
            'images': images,
            'like': 0,
            'link': link,
            'oldprice': oldprice,
            'pp': 'advcake',
            'price': price,
            'sale': sale,
            "season": False,
            'season_t': False,
            'shop': 'brandshop',
            'sizes': sizes,
            "style": False,
            'style_t': False,
            'structure': structure,
            'subcategory': subcategory,
            'subcategory_t': get_transliterate(subcategory)
        }
    except:
        print(f'{link} НЕ собран!')
        return

File: admitad/elyts_proto/getProduct.js
const axios = require('axios')
const { JSDOM } = require('jsdom')
const getOptions = require('./getOptions')
const getTransliterate = require('./getTransliterate')


module.exports = async function getProduct (gender, productLink) {
  try {
    // получаю данные по товару
    const productOptions = getOptions(null, productLink)
    const { data } = await axios.get( ...productOptions )
    const dom = new JSDOM(data).window.document
    const article = dom.querySelector('span[itemprop="mpn"]').textContent
    let available = +dom.querySelector('.goods-remainder').textContent.split(' ')[0]
    if (isNaN(available)) available = 1
    let brand = dom.querySelector('span[itemprop="brand"]').textContent
    if (brand === 'P.A.R.O.S.H.') brand = 'PAROSH'
    const breadCrumbs = Array.from(dom.querySelectorAll('span[property="name"]'))
      .map(crumb => crumb.textContent)
    const category = breadCrumbs[0]
    let subcategory = breadCrumbs[breadCrumbs.length - 1]
    if (subcategory.includes('ые') || subcategory.includes('сс'))
      subcategory = breadCrumbs[breadCrumbs.length - 2]

    const color = dom.querySelector('.b-product__color__value').textContent.replaceAll('ё', 'е').toLowerCase()
    let description
    try { description = dom.querySelector('#goods_description').textContent }
    catch (e) { description = false }
    const cashback = +dom.querySelector('.b-cashback').querySelector('span').textContent.replace(' ', '').replace('руб.', '')
    const oldprice = +dom.querySelector('.price-item').querySelector('del').textContent.trim().replaceAll(' ', '').replace('руб.', '')
    const price = +dom.querySelector('.final-price').textContent.trim().replace(' ', '').replace('руб.', '')
    const sale = +dom.querySelector('.discount-percent').textContent.replace('(-', '').replace('%)', '')
    let name = dom.querySelector('h1[itemprop="name"]').textContent
    name = name.includes(', ') ? name.split(', ')[0] : name
    const images = Array.from(dom.querySelectorAll('.picture-item img')).map(img => img.getAttribute('data-original'))
    const sizes = Array.from(
      Array.from(dom.querySelectorAll('.b-size-select__pane')
      )[0].querySelectorAll('option')).map(size => size.textContent
      .replace(' (в резерве)', '').replace(' / ', '-').replace(' INT', ''))

    const info = {}
    let country = false
    let structure = false
    const params = Array.from(dom.querySelectorAll('.b-title')).map(tag => tag.textContent.trim())
    const values = Array.from(dom.querySelectorAll('.b-value')).map(tag => tag.textContent.trim())
    params.forEach((param, idx) => info[param.replace(':', '')] = values[idx])
    if ('Страна производитель' in info) country = info['Страна производитель']; delete info['Страна производитель']
    if ('Состав' in info) structure = info['Состав']; delete info['Состав']
    return {
      id: Math.floor(Math.random() * 9e9) + 1e9,
      age: 'Взрослый',
      article,
      available,
      benefit: oldprice - price,
      bonus: { cashback },
      brand,
      brandCountry: false,
      brandCountry_t: false,
      category,
      category_t: getTransliterate(category),
      color,
      color_t: color ? getTransliterate(color) : color,
      country,
      country_t: country ? getTransliterate(country) : country,
      create: new Date,
      delivery: [ 'ru', 'kz' ],
      deliveryPrice: 500,
      description,
      gender,
      installment: false,
      images,
      like: 0,
      link: productLink,
      name,
      shop: 'elyts',
      info,
      oldprice,
      pp: 'admitad',
      price,
      sale,
      season: false,
      season_t: false,
      sizes,
      style: false,
      style_t: false,
      structure,
      subcategory,
      subcategory_t: getTransliterate(subcategory).replaceAll(' ', '-')
    }
  } catch (e) {
    console.log(e)
    console.log('Товар', productLink, 'не собран')
    return null
  }
}
